shared/src/main/scala/tastyquery/ast/Names.scala

```diff
@@ -80,15 +86,20 @@ object Names {
   abstract class TermName extends Name {
     def tag: Int
 
-    private var myTypeName: TypeName = null
+    private var myTypeName: TypeName | Null = null
 
     override def toTypeName: TypeName = {
-      if (myTypeName == null) {
-        synchronized {
-          if (myTypeName == null) myTypeName = new TypeName(this)
-        }
+      val local = myTypeName
+      if local == null then synchronized {
+        val local1 = myTypeName
```

Comment:
Consider calling them `local1` and `local2`, since neither is really more important than the other.

Also, consider putting the `synchronized` block on the next line, otherwise it appears you're mixing `if then` with `{}`, which is a bit confusing.